File: Introduction/c-lang/speed-loop.c
// There's no code in the body of the for loop, but your processor is busy incrementing i, comparing it to cycles, and repeating the process. It's late 2017 as I write this, typing on a MacBook Pro with a 2.8 GHz Intel Core i7 processor. Node v9.3.0 is current, and takes less than a second to loop a billion times. How fast is pure C? Let's see.
#include <stdio.h>
#include <time.h>
int main() {
int cycles = 1000000000;
clock_t start, end;
double duration;
start = clock();
for (int i = 0; i < cycles; i++) {
/* Empty loop */
}
end = clock();
duration = ((double)(end - start)) / CLOCKS_PER_SEC;
printf("C looped %d times in %lf seconds\n", cycles,duration);
return 0;
}

File: chapter 1/Introduction/python/speed-loop.py
# For additional comparison, let's try an interpreted language, like Python.

import time
def main():
cycles = 1000000000
start = time.perf_counter()
for i in range(0, cycles):
pass # Empty loop
end = time.perf_counter()
duration = end - start
print("Python looped %d times in %.3f seconds" % (cycles, duration))
main()
